src/Network/AWS/ARN/S3.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_HADDOCK show-extensions #-}
+
+-- |
+--
+-- Module      : Network.AWS.ARN.S3
+-- Copyright   : (C) 2020-2022 Bellroy Pty Ltd
+-- License     : BSD-3-Clause
+-- Maintainer  : Bellroy Tech Team <haskell@bellroy.com>
+-- Stability   : experimental
+module Network.AWS.ARN.S3
+  ( -- * S3 Resource
+    S3Resource (..),
+    parseS3Resource,
+    renderS3Resource,
+
+    -- ** Prisms
+    _S3Resource,
+  )
+where
+
+import Data.Foldable (fold)
+import Data.Hashable (Hashable)
+import Data.Text (Text)
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import Network.AWS.ARN.Internal.Lens (Prism', prism')
+
+-- $setup
+-- >>> :set -XOverloadedStrings
+-- >>> import Network.AWS.ARN.Internal.Lens ((^?))
+
+-- | An AWS S3 bucket, and optional object key.
+--
+-- >>> "bucket-name" ^? _S3Resource
+-- Just (S3Resource {bucketName = "bucket-name", objectKey = Nothing})
+--
+-- >>> "bucket-name/my/object" ^? _S3Resource
+-- Just (S3Resource {bucketName = "bucket-name", objectKey = Just "/my/object"})
```

Comment:
The `/` in the ARN is a delimiter, not part of the key.
https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html

```suggestion
-- >>> "bucket-name/my/object" ^? _S3Resource
-- Just (S3Resource {bucketName = "bucket-name", objectKey = Just "my/object"})
```

(This means that `parseS3Resource` should remove `/`, and `renderS3Resource` should insert `/`)

Reply:
Oh, good catch, will fix